app/controllers/projects_controller.rb

```diff
@@ -36,6 +36,22 @@ def show
     commontator_thread_show(@project)
   end
 
+  def project_invite
+    if Project.with_project_token.exists?(collaboration_token: params[:token])
+      if current_user.collaborations.exists?(project: @project)
```

Comment:
When user is unauthenticated, current_user is nil. So better use `current_user&.collaborations....`